pgx/savepoint.go

```diff
@@ -0,0 +1,37 @@
+package pgx
+
+//go:generate mockgen -source=$GOFILE -destination=mock/$GOFILE -package=mock
+
+import "fmt"
+
+// This file copies an interface from https://github.com/DATA-DOG/go-txdb
+
+// SavePoint defines the syntax to create savepoints
+// within transaction.
+type SavePoint interface {
+	Create(id string) string
+	Release(id string) string
+	Rollback(id string) string
+}
+
+//revive:disable:exported
+type DefaultSavePoint struct{}
+
+// NewSavePoint is an implementation of trm.SavePoint.
+func NewSavePoint() DefaultSavePoint {
```

Comment:
Could it be reused from trm.sql package?

Reply:
done